Tensile/KernelWriterAssembly.py

```diff
@@ -7684,15 +7696,29 @@ def globalReadGuardK(self, kernel, tP, vregSetIdx):
                   offsetVgpr = self.guardZeroPad(kernel, tP, codeMod, offsetVgpr, soffset, addrV, perp, sPerp, para, sPara)
                   kStr += str(codeMod)
 
-                unrollMirrorWithSoffset = kernel["ProblemType"]["IndicesSummation"][self.unrollIdx] in problemType["MirrorDims%s"%tc] and soffset != "0"
-                # ScalarGlobalReadOffset should be negative value with unroll mirroring.
-                # However, buffer_load uses soffset as uint value, so GRO - SGRO, SGRO = 0
-                if unrollMirrorWithSoffset:
+                if mirrorDim:
                   codeMod = Code.Module("mirrorIdx%u"%loopCnt)
-                  codeMod.addInst("_v_sub_u32", vgpr(offsetVgpr), vgpr(offsetVgpr), soffset, "mirror unroll: GRO=GRO-SGRO, soffset=0")
+                  codeMod.addInst("v_mov_b32", vgpr(mirrorOffsetTemp), vgpr(offsetVgpr), "copy offset to clamp mirror dim")
+                  offsetVgpr = mirrorOffsetTemp
+
+                  # ScalarGlobalReadOffset should be negative value with unroll mirroring.
+                  # However, buffer_load uses soffset as uint value, so GRO - SGRO, SGRO = 0
+                  if soffset != "0":
+                    codeMod.addInst("_v_sub_u32", vgpr(offsetVgpr), vgpr(offsetVgpr), soffset, "mirror unroll: GRO=GRO-SGRO, soffset=0")
+                    # soffset_prev = soffset
+                    soffset = "0"
+
+                  # check for pre-padding
+                  prePad = 0
+                  if self.groOffsetInMacroTile:
+                    prePad = self.srdShiftLeft[tc]
+                    if tc == "A":
```

Comment:
We can simply use self.bpeAB here. No need to put if/elif here.

Reply:
Updated